OgmentoAPI.Domain.Catalog.Abstractions/Dto/ProductDtoMapping.cs

```diff
@@ -0,0 +1,42 @@
+﻿using OgmentoAPI.Domain.Catalog.Abstractions.Models;
+using OgmentoAPI.Domain.Common.Abstractions.Dto;
+
+namespace OgmentoAPI.Domain.Catalog.Abstractions.Dto
+{
+	public static class ProductDtoMapping
+	{
+		public static ProductDto ToDto(this ProductModel product)
+		{
+			ProductDto productDto = new ProductDto()
+			{
+				ProductName = product.ProductName,
+				ProductDescription = product.ProductDescription,
+				Price = product.Price,
+				Weight = product.Weight,
+				LoyaltyPoints = product.LoyaltyPoints,
+				ProductExpiry = product.ProductExpiry,
+				SkuCode = product.SkuCode,
+				Categories = product.Categories.Select(CategoryDtoMapping.ToDto).ToList(),
+				Images = product.Images != null ? product.Images.Select(PictureDtoMapping.ToDto).ToList() : [],
+			};
```

Comment:
```suggestion
		=> new ProductDto()
			{
				ProductName = product.ProductName,
				ProductDescription = product.ProductDescription,
				Price = product.Price,
				Weight = product.Weight,
				LoyaltyPoints = product.LoyaltyPoints,
				ProductExpiry = product.ProductExpiry,
				SkuCode = product.SkuCode,
				Categories = product.Categories.Select(CategoryDtoMapping.ToDto).ToList(),
				Images = product.Images != null ? product.Images.Select(PictureDtoMapping.ToDto).ToList() : [],
			};
			
```
You can directly return the model and save creating one object in memory